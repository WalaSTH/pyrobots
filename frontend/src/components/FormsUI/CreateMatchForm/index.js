import { Card, Grid, Typography, Box } from "@mui/material";
import TextField from "../TextField";
import SubmitFormButton from "../SubmitFormButton";
import SelectRobot from "../SelectRobot";
import { Formik, Form } from "formik";
import * as Yup from "yup";

const formValidation = Yup.object().shape({
  name: Yup.string()
    .min(3, "Match name must have more than 3 characters")
    .max(20, "Match name cant contain more than 20 characters")
    .matches(
      "^[A-Za-z0-9 ]*$",
      "Match name can only contains letters, numbers or spaces"
    ),
  password: Yup.string()
    .matches(
      "^[A-Za-z0-9 ]*$",
      "Password can only contains letters, numbers or spaces"
    )
    .max(64, "Password cant contain more than 64 characters")
    .notRequired(),
  min_players: Yup.number()
    .integer()
    .typeError("Insert a number")
    .positive()
    .min(2, "Min players must be greater or equal than 2")
    .max(4, "Min players must be lower or equal than 4"),
  max_players: Yup.number()
    .integer()
    .typeError("Insert a number")
    .positive()
    .min(
      Yup.ref("min_players"),
      "Max players must be greater or equal than Min players"
    )
    .max(4),
  games_per_match: Yup.number()
    .integer()
    .typeError("Insert a number")
    .positive()
    .max(200),
  rounds: Yup.number()
    .integer()
    .typeError("Insert a number")
    .positive()
    .max(10000),
  robot_id: Yup.number().integer().positive().required(),
});

export default function CreateMatchForm({ userID, handleSubmit }) {
  const initialFormState = {
    name: "",
    password: "",
    min_players: "",
    max_players: "",
    games_per_match: "",
    rounds: "",
    robot_id: "",
    creator: userID,
  };

  return (
    <Box
      sx={{
        marginTop: 10,
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
      }}
    >
      <Formik
        initialValues={{
          ...initialFormState,
        }}
        validationSchema={formValidation}
        onSubmit={handleSubmit}
      >
        <Form>
          <Card
            variant="outlined"
            sx={{ marginTop: 3, padding: 3, borderRadius: 3 }}
          >
            <Grid container spacing={2}>
              <Grid item xs={12} sx={{ textAlign: "center" }}>
                <Typography component="h1" variant="h5">
                  Create Match
                </Typography>
              </Grid>

              <Grid item xs={12}>
                <TextField
                  name="name"
                  label="Name of the match"
                  autoComplete="off"
                  required
                />
              </Grid>

              <Grid item xs={12}>
                <TextField
                  name="password"
                  label="Password (Optional)"
                  type="password"
                  data-testid="passwordInput"
                  autoComplete="off"
                />
              </Grid>

              <Grid item xs={6} textAlign="center">
                <TextField
                  name="min_players"
                  label="Min players"
                  placeholder="2"
                  autoComplete="off"
                  required
                />
              </Grid>

              <Grid item xs={6} textAlign="center">
                <TextField
                  name="max_players"
                  label="Max players"
                  placeholder="4"
                  autoComplete="off"
                  required
                />
              </Grid>

              <Grid item xs={6}>
                <TextField
                  name="games_per_match"
                  label="Games"
                  placeholder="200"
                  autoComplete="off"
                  required
                />
              </Grid>

              <Grid item xs={6}>
                <TextField
                  name="rounds"
                  label="Rounds"
                  placeholder="10000"
                  autoComplete="off"
                  required
                />
              </Grid>

              <Grid item xs={12} textAlign="center">
                <SelectRobot name="robot_id" getRobotName="0" />
              </Grid>

              <Grid item xs={12}>
                <SubmitFormButton data-testid="createMatchButton">
                  Create match
                </SubmitFormButton>
              </Grid>
            </Grid>
          </Card>
        </Form>
      </Formik>
    </Box>
  );
}
