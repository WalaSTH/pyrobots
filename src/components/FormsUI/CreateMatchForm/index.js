import { Form } from "formik";
import { Card, Grid, Typography } from "@mui/material";
import TextField from "../TextField";
import SubmitFormButton from "../SubmitFormButton";

export default function CreateMatchForm() {
  return <div></div>;
}
//   return (
//     <Box
//       sx={{
//         marginTop: 10,
//         display: "flex",
//         flexDirection: "column",
//         alignItems: "center",
//       }}
//     >
//       <Formik
//         initialValues={{
//           ...initialFormState,
//         }}
//         validationSchema={formValidation}
//         onSubmit={handleSubmit}
//       >
//         <Form>
//           <Card
//             variant="outlined"
//             sx={{ marginTop: 3, padding: 3, borderRadius: 3 }}
//           >
//             <Grid container spacing={2}>
//               <Grid item xs={12} sx={{ textAlign: "center" }}>
//                 <Typography sx={{ fontSize: 18, fontWeight: 500 }}>
//                   Create Match
//                 </Typography>
//               </Grid>

//             <Grid item xs={12}>
//               <TextField
//                 name="name"
//                 label="Name of the match"
//                 autoComplete="off"
//               />
//             </Grid>

//             <Grid item xs={12}>
//               <TextField
//                 name="password"
//                 label="Password (Optional)"
//                 type="password"
//                 autoComplete="off"
//               />
//             </Grid>

//             <Grid item xs={6} textAlign="center">
//               <TextField
//                 name="min_players"
//                 label="Min players"
//                 placeholder="2"
//                 autoComplete="off"
//               />
//             </Grid>

//             <Grid item xs={6} textAlign="center">
//               <TextField
//                 name="max_players"
//                 label="Max players"
//                 placeholder="4"
//                 autoComplete="off"
//               />
//             </Grid>

//             <Grid item xs={6}>
//               <TextField
//                 name="games_per_match"
//                 label="Games"
//                 placeholder="200"
//                 autoComplete="off"
//               />
//             </Grid>

//             <Grid item xs={6}>
//               <TextField
//                 name="rounds"
//                 label="Rounds"
//                 placeholder="10000"
//                 autoComplete="off"
//               />
//             </Grid>

//             <Grid item xs={12} textAlign="center">
//               <Typography>Choose your robot!</Typography>
//               <TextField name="robot_id" label="Robot ID" autoComplete="off" />
//             </Grid>

//             <Grid item xs={12}>
//               <SubmitFormButton>Create match</SubmitFormButton>
//             </Grid>
//           </Grid>
//         </Card>
//       </Form>
//     </>
//   );
// }
