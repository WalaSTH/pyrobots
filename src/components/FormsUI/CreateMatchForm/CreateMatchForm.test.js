import React from "react";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import CreateMatchForm from "./";

describe("<CreateMatchForm", () => {
  const user = userEvent.setup();
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("component rendering", async () => {
    const component = renderer.create(<CreateMatchForm />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("component renders with all the components", () => {
    render(<CreateMatchForm />);
    screen.getByLabelText(/Name of the match/i);
    screen.getByLabelText(/Password/i);
    screen.getByLabelText(/Min players/i);
    screen.getByLabelText(/Max players/i);
    screen.getByLabelText(/Games/i);
    screen.getByLabelText(/Rounds/i);
    screen.getByLabelText(/Robot ID/i);
  });

  it("validates fields to have something after write", async () => {
    CreateMatchForm.handleSubmit = jest.fn();
    render(<CreateMatchForm />);

    const nameOfTheMatchInput = screen.getByLabelText(/Name of the match/i);
    const minPlayersInput = screen.getByLabelText(/Password/i);
    const maxPlayersInput = screen.getByLabelText(/Min players/i);
    const gamesInput = screen.getByLabelText(/Max players/i);
    const roundsInput = screen.getByLabelText(/Games/i);
    const robotIdInput = screen.getByLabelText(/Rounds/i);

    await user.type(nameOfTheMatchInput, "diego");
    await user.type(minPlayersInput, "2");
    await user.type(maxPlayersInput, "4");
    await user.type(gamesInput, "200");
    await user.type(roundsInput, "10000");
    await user.type(robotIdInput, "1");

    const button = screen.getByRole("button", { name: /Create-match/i });
    await user.click(button);

    expect(nameOfTheMatchInput.value).toBe("diego");
    expect(minPlayersInput.value).toBe("2");
    expect(maxPlayersInput.value).toBe("4");
    expect(gamesInput.value).toBe("200");
    expect(roundsInput.value).toBe("10000");
    expect(robotIdInput.value).toBe("1");
  });

  it("mock fetch", async () => {
    const fakeMatch = {
      name: "PatriaoPaper",
      password: "",
      min_players: "2",
      max_players: "4",
      games_per_match: "10000",
      rounds: "200",
      robot_id: "1",
      creator: 1,
    };

    await CreateMatchForm.handleSubmit(fakeMatch);
    expect(CreateMatchForm.handleSubmit).toHaveBeenCalledTimes(1);
    expect(CreateMatchForm.handleSubmit).toHaveBeenCalledWith(fakeMatch);
  });
});
