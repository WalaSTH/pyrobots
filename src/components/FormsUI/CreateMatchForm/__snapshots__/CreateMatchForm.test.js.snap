// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateMatchForm component rendering 1`] = `
<div
  className="MuiBox-root css-nosrnw"
>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-xheubk-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1fpckdt-MuiGrid-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-1xh9tsm-MuiTypography-root"
          >
            Create Match
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r0:"
              id=":r0:-label"
            >
              Name of the match
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r0:"
                name="name"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Name of the match
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            data-testid="passwordInput"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r1:"
              id=":r1:-label"
            >
              Password (Optional)
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r1:"
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Password (Optional)
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-v0rsyg-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r2:"
              id=":r2:-label"
            >
              Min players
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r2:"
                name="min_players"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="2"
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Min players
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-v0rsyg-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r3:"
              id=":r3:-label"
            >
              Max players
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r3:"
                name="max_players"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="4"
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Max players
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r4:"
              id=":r4:-label"
            >
              Games
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r4:"
                name="games_per_match"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="200"
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Games
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r5:"
              id=":r5:-label"
            >
              Rounds
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r5:"
                name="rounds"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="10000"
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Rounds
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1fpckdt-MuiGrid-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Choose your robot!
          </p>
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r6:"
              id=":r6:-label"
            >
              Robot ID
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r6:"
                name="robot_id"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Robot ID
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <button
            aria-label="Create-match"
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-189e9pj-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            Create match
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
